flytekit/configuration/__init__.py

```diff
@@ -237,18 +237,13 @@ def auto(cls, config_file: typing.Union[str, ConfigFile] = None, img_name: Optio
         :param img_name:
         :return:
         """
-        if config_file is None and img_name is None:
-            raise ValueError("Either an image or a config with a default image should be provided")
```

Comment:
I think we need to keep this right?  Why get rid of this?  What should happen if both are None?

Reply:
If config file is None, we should use [get_config_file](https://github.com/flyteorg/flytekit/blob/195d26ab0cc03260bb96b828219fcf7e6ea768a1/flytekit/configuration/file.py#L231) to read config from default path or from env variables

Reply:
got it yeah, it was returning None earlier, fixed that.